lib/hex_web/validation.ex

```diff
@@ -52,27 +17,14 @@ defmodule HexWeb.Validation do
     end)
   end
 
-  def validate_meta(changeset, field, meta_types, meta_required) do
-    validate_change(changeset, field, fn _field, meta ->
-      type_errors =
-        Enum.flat_map(meta_types, fn {sub_field, type} ->
-          type(sub_field, Map.get(meta, sub_field), type)
-        end)
-
-      req_errors =
-        Enum.flat_map(meta_required, fn field ->
-          if Map.has_key?(meta, field) do
-            []
-          else
-            [{field, :missing}]
-          end
-        end)
-
-      errors = req_errors ++ type_errors
-
-      if errors == [],
-          do: [],
-        else: [{field, errors}]
-    end)
+  @doc """
+  Adds embed errors to the changeset.errors
+  """
+  def put_embed_errors(%Ecto.Changeset{changes: changes} = changeset, embed) do
```

Comment:
The idea is to keep all errors in the changesets and use the `Ecto.Changeset.traverse_errors` functions to traverse them into a single list.